hw/xbox/dsp/dsp_cpu.c

```diff
@@ -420,23 +431,29 @@ void dsp56k_reset_cpu(dsp_core_t* dsp)
     dsp->disasm_prev_inst_pc = 0xFFFFFFFF;
 }
 
-static OpcodeEntry lookup_opcode(uint32_t op) {
+static OpcodeEntry lookup_opcode_manually(uint32_t op)
+{
     OpcodeEntry r = {0};
     int i;
     for (i=0; i<ARRAYSIZE(nonparallel_opcodes); i++) {
         if ((op & nonparallel_matches[i][0]) == nonparallel_matches[i][1]) {
             if (nonparallel_opcodes[i].match_func 
                 && !nonparallel_opcodes[i].match_func(op)) continue;
             if (r.template != NULL) {
-                printf("qqq %x %s\n", op, r.template);
+                //printf("qqq %x %s\n", op, r.template);
             }
-            assert(r.template == NULL);
+            //assert(r.template == NULL);
             r = nonparallel_opcodes[i];
         }
     }
     return r;
 }
 
+static OpcodeEntry lookup_opcode(uint32_t op)
+{
+    return OpcodeEntryArray[op];
+}
```

Comment:
With the changed design it might not be necessary to have this as part of a function, so it could be inline probably?

Reply:
Yeah :thumbsup:

Reply:
:+1:  done